src/execution_plan/execution_plan_build/build_projection_ops.c

```diff
@@ -112,6 +113,12 @@ static void _combine_projection_arrays(AR_ExpNode ***exps_ptr, AR_ExpNode **orde
 		raxTryInsert(projection_names, (unsigned char *)name, strlen(name), NULL, NULL);
 	}
 
+	// check if the projection expressions contain aggregation an expression
+	bool is_agg = false;
+	for(uint i = 0; i < project_count && !is_agg; i++) {
+		is_agg = AR_EXP_ContainsAggregation(project_exps[i]);
+	}
+
```

Comment:
Why do we need this second loop,
simply perform this assignment on the previous loop,
also update the assignment to: `is_agg |= AR_EXP_ContainsAggregation(project_exps[i]);`